From 5e6907fe0603b5f1aa387288e275b4c685af722f Mon Sep 17 00:00:00 2001
From: Olivier Moysan <olivier.moysan@foss.st.com>
Date: Fri, 28 Jul 2023 10:24:18 +0200
Subject: [PATCH 0494/1141] arm64: dts: st: enable hdmi audio on
 stm32mp257f-ev1

Add HDMI audio support through ADV7535 HDMI transceiver on
STM32MP257F-EV1 board.
The sound card and i2s2 nodes are disabled by default.
If the HDMI MB1752 board is detected, these nodes are enabled
in uboot at runtime.

Signed-off-by: Olivier Moysan <olivier.moysan@foss.st.com>
Change-Id: Ie408d8d57d5543f12c03e6557730e7fef6e94c5b
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/324538
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp257f-ev1.dts | 41 ++++++++++++++++++++--
 1 file changed, 38 insertions(+), 3 deletions(-)

--- a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
@@ -117,6 +117,13 @@
 		default-brightness-level = <0>;
 		status = "okay";
 	};
+
+	sound: sound {
+		compatible = "audio-graph-card";
+		label = "STM32MP25-EV1";
+		dais = <&i2s2_port>;
+		status = "disabled";
+	};
 };
 
 &a35ss_syscfg {
@@ -342,9 +349,22 @@
 		v3p3-supply = <&scmi_v3v3>;
 		v1p2-supply = <&scmi_v3v3>;
 
-		port {
-			adv7535_in: endpoint {
-				remote-endpoint = <&dsi_out2>;
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			port@0 {
+				reg = <0>;
+				adv7535_in: endpoint {
+					remote-endpoint = <&dsi_out2>;
+				};
+			};
+
+			port@2 {
+				reg = <2>;
+				adv7535_tx_endpoint: endpoint {
+					remote-endpoint = <&i2s2_endpoint>;
+				};
 			};
 		};
 	};
@@ -370,6 +390,21 @@
 	};
 };
 
+&i2s2 {
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&i2s2_pins_a>;
+	pinctrl-1 = <&i2s2_sleep_pins_a>;
+	status = "disabled";
+
+	i2s2_port: port {
+		i2s2_endpoint: endpoint {
+			remote-endpoint = <&adv7535_tx_endpoint>;
+			format = "i2s";
+			mclk-fs = <256>;
+		};
+	};
+};
+
 &ipcc1 {
 	status = "okay";
 };
